pkg/controllers/csi/metadata/migrations.go

```diff
@@ -0,0 +1,100 @@
+package metadata
+
+import (
+	dtcsi "github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi"
+	"gorm.io/gorm"
+)
+
+func dataMigration(tx *gorm.DB) error {
+	var dynakubes []Dynakube
+
+	tx.Table("dynakubes").Find(&dynakubes)
+
+	pr := PathResolver{RootDir: dtcsi.DataPath}
+
+	for _, d := range dynakubes {
+		var version string
+		if d.LatestVersion != "" {
+			version = d.LatestVersion
+		} else if d.ImageDigest != "" {
+			version = d.ImageDigest
+		}
+
+		tc := TenantConfig{
+			Name:                        d.Name,
+			TenantUUID:                  d.TenantUUID,
+			ConfigDirPath:               pr.AgentConfigDir(d.TenantUUID, d.Name),
+			DownloadedCodeModuleVersion: version,
+			MaxFailedMountAttempts:      int64(d.MaxFailedMountAttempts),
+		}
+
+		result := tx.Create(&tc)
+		if result.Error != nil {
+			log.Error(result.Error, "failed to create TenantConfig")
+		}
+
+		cm := CodeModule{
+			Version:  version,
+			Location: pr.AgentSharedBinaryDirBase(),
```

Comment:
It should be [AgentSharedBinaryDirForAgent](https://github.com/Dynatrace/dynatrace-operator/blob/784e35fc06bca116c3cbdd2d6f775703729f5972/pkg/controllers/csi/metadata/path_resolver.go#L43-L43) as you want to exact location

Reply:
fixed cbdf7da2